commit e5767eaad1c48a23eb3d8fed4616722189a70aa1
Author: Sergey Sharybin <sergey.vfx@gmail.com>
Date:   Mon Jul 2 15:11:48 2018 +0200

    Cycles: Fix missing curve hair when building with GCC-8 in release mode
    
    Reshuffle cast intrinsics to make XOR to operate on __m128i rather
    than on __m128.
    
    Hopefully this does not affect performance.

diff --git a/intern/cycles/util/util_ssef.h b/intern/cycles/util/util_ssef.h
index bb007ff84a9..e9a617bb2cc 100644
--- a/intern/cycles/util/util_ssef.h
+++ b/intern/cycles/util/util_ssef.h
@@ -606,7 +606,7 @@ ccl_device_inline const ssef uint32_to_float(const ssei &in)
 template<size_t S1, size_t S2, size_t S3, size_t S4>
 ccl_device_inline const ssef set_sign_bit(const ssef &a)
 {
-	return a ^ cast(ssei(S1 << 31, S2 << 31, S3 << 31, S4 << 31));
+	return cast(cast(a) ^ ssei(S1 << 31, S2 << 31, S3 << 31, S4 << 31));
 }
 
 ////////////////////////////////////////////////////////////////////////////////
